packages/eslint-plugin/tests/rules/strict-boolean-expressions.test.ts

```diff
@@ -133,28 +133,44 @@ ruleTester.run('strict-boolean-expressions', rule, {
       `,
     }),
 
-    // nullable enum in boolean context
+    // enum in boolean context
+    ...batchedSingleLineTests<Options>({
+      code: noFormat`
+        enum Foo { Bar, Baz }; (val: Foo) => val ? 1 : 0;
+        enum Foo { Bar, Baz = 1 }; (val: Foo) => val ? 1 : 0;
+        enum Foo { Bar = 0, Baz = 1 }; (val: Foo) => val ? 1 : 0;
+        enum Foo { Bar = 1, Baz = 2 }; (val: Foo) => val ? 1 : 0;
+        enum Foo { Bar, Baz = 'baz' }; (val: Foo) => val ? 1 : 0;
+        enum Foo { Bar = '', Baz = 'baz' }; (val: Foo) => val ? 1 : 0;
+        enum Foo { Bar = 'bar', Baz = 'baz' }; (val: Foo) => val ? 1 : 0;
+        enum Foo { Bar = 0, Baz = 'baz' }; (val: Foo) => val ? 1 : 0;
+        enum Foo { Bar = 1, Baz = '' }; (val: Foo) => val ? 1 : 0;
+        enum Foo { Bar = 0, Baz = '' }; (val: Foo) => val ? 1 : 0;
+        enum Foo { Bar = 1, Baz = 'baz' }; (val: Foo) => val ? 1 : 0;
+      `,
+      options: [{ allowEnum: true }],
+    }),
     {
       code: `
         enum ExampleEnum {
-          This = 0,
-          That = 1,
+          This = 1,
+          That = 2,
         }
         const rand = Math.random();
         let theEnum: ExampleEnum | null = null;
         if (rand < 0.3) {
           theEnum = ExampleEnum.This;
         }
-        if (theEnum) {
+        if (!theEnum) {
         }
       `,
-      options: [{ allowNullableEnum: true }],
+      options: [{ allowEnum: true }],
     },
     {
       code: `
         enum ExampleEnum {
-          This = 0,
-          That = 1,
+          This = 'one',
+          That = 'two',
```

Comment:
This diff looks weird. I've moved the tests for `is('nullish', 'truthy number', 'enum')` and `is('nullish', 'truthy string', 'enum')` conditions to the `allowEnum` tests group and I've changed the option to be `allowEnum` instead of `allowNullableEnum`